apps/anoma_node/lib/examples/e_shard_backend.ex

```diff
@@ -0,0 +1,718 @@
+    :ok = Mempool.tx(node_id, {:shard_storage, tx3_code}, "tx3")
+
+    # Ensure all submitted transactions are present before execution
+    tx_ids_dump = Mempool.tx_dump(node_id)
+    ordered_tx_ids = ["tx1", "tx2", "tx3"]
+    assert MapSet.new(tx_ids_dump) == MapSet.new(ordered_tx_ids)
+
+    # 4. Execute All Transactions Together
+    Logger.info("Executing Tx1, Tx2, Tx3 concurrently...")
+    :ok = Mempool.execute(node_id, ordered_tx_ids)
+    :ok = wait_for_block(node_id, 1)
+    Logger.info("Finished executing Tx1, Tx2, Tx3.")
+    log_shard_states(node_id, "After Concurrent Execution")
+
+    # Wait for the final watermarks to be updated by Ordering
+    wait_for_state_condition(
+      Registry.whereis(node_id, Shard, :a),
+      "read WM on a == 3 after concurrent execution",
+      fn state ->
+        get_in(state.watermarks, ["a", :read]) == 3
+      end
+    )
+
+    wait_for_state_condition(
+      Registry.whereis(node_id, Shard, :b),
+      "read WM on b == 3 after concurrent execution",
+      fn state ->
+        get_in(state.watermarks, ["b", :read]) == 3
+      end
+    )
+
+    wait_for_state_condition(
+      Registry.whereis(node_id, Shard, :c),
+      "write WM on c == 3 after concurrent execution",
+      fn state ->
+        get_in(state.watermarks, ["c", :write]) == 3
+      end
+    )
+
+    # 5. Verification (Final state only)
+    pid_a = Registry.whereis(node_id, Shard, :a)
+    pid_b = Registry.whereis(node_id, Shard, :b)
+    pid_c = Registry.whereis(node_id, Shard, :c)
+
+    assert is_pid(pid_a), "Shard 'a' PID not found."
+    assert is_pid(pid_b), "Shard 'b' PID not found."
+    assert is_pid(pid_c), "Shard 'c' PID not found."
+
+    # Check final state using :sys.get_state
+    state_a = :sys.get_state(pid_a)
+    state_b = :sys.get_state(pid_b)
+    state_c = :sys.get_state(pid_c)
+
+    Logger.debug("Shard A final state: #{inspect(state_a)}")
+    Logger.debug("Shard B final state: #{inspect(state_b)}")
+    Logger.debug("Shard C final state: #{inspect(state_c)}")
+
+    # Verify final KV values (Heights reflect execution order)
+    assert state_a.kv["a"][1].value == 3,
+           "Shard 'a' state mismatch for height 1 (Tx1)"
+
+    assert state_b.kv["b"][2].value == 4,
+           "Shard 'b' state mismatch for height 2 (Tx2)"
+
+    assert state_c.kv["c"][3].value == 7,
+           "Shard 'c' state mismatch for height 3 (Tx3)"
+
+    # Verify final watermarks (Reflect all reads/writes across the executed block)
+    assert state_a.watermarks["a"].read == 3,
+           "Shard 'a' read watermark should be 3 after concurrent execution"
+
+    assert state_b.watermarks["b"].read == 3,
+           "Shard 'b' read watermark should be 3 after concurrent execution"
+
+    # Only Tx3 wrote to 'c'
+    assert state_c.watermarks["c"].write == 3,
+           "Shard 'c' write watermark should be 3 after concurrent execution"
+
+    # 6. Cleanup
+    :ok = ENode.stop_node(enode)
+    :ok
```

Comment:
Why are we cleaning up? Why not let the node live as an argument to the example, see transaction examples that have the optional argument to shut down.

Basically we wish to carry on with this in further examples that you or others can write